### PR TITLE
Fix SIM111 warning for compound expressions.

### LINKED ISSUE #59: [Adjust Rule] SIM111 for compound conditionals
## Desired change

* **Rule(s)**: SIM111
* **Adjustment**: Fix warning message for compound conditionals

## Explanation

The warning message is incorrect for compound expressions.  It can be fixed by enclosing the conditional in parentheses.

## Example

This is an example where the mentioned rule(s) would currently be ~~suboptimal~~ wrong:

```python
for x in iterable:
    if A(x) and B(x):
        return False
return True
```
Result, with correction:
```diff
- SIM111: Use 'return all(not A(x) and B(x) for x in iterable)'
+ SIM111: Use 'return all(not (A(x) and B(x)) for x in iterable)'
                              ^             ^
```
or (but you probably don't want to be parsing the expression and doing boolean algebra?):
```diff
- SIM111: Use 'return all(not A(x) and B(x) for x in iterable)'
+ SIM111: Use 'return all(not A(x) or not B(x) for x in iterable)'
```

### PR DESCRIPTION
**Pull request by hjoliver**
Close #59

### CODE REVIEW

**Comment by hjoliver** [comment]
Happy to extend the tests if you agree with this change.

File: flake8_simplify.py [hunk 1]
```diff
@@ -598,9 +598,21 @@ def _get_sim110_sim111(node: ast.For) -> List[Tuple[int, int, str]]:
...
     elif node.body[0].body[0].value.value is False:
-        check = "not " + check
-        if check.startswith("not not"):
-            check = check[len("not not ") :]
+        if (
+            " and " not in check
+            and " or " not in check
+        ):
+            # Simple (non-compound) case
+            if check.startswith("not "):
```
Comment (Skylion007) [inline flake8_simplify.py:606]: You should be checking with this with the ast.

**Comment by MartinThoma** [comment]
Hi @hjoliver , thank you for the patch!

I'd like a tiny stylistic change when I merge it:

```
is_compound_expression = " and " in check or " or " in check

if is_compound_expression:
    check = f"not ({check})"
else:
    if check.startswith("not "):
        # Remove the existing "not " to negate the check
        check = check[len("not "):]
    else:
        # Add "not " to negate the check
        check = f"not {check}"
```

I could also add it to the PR and squash merge if that is fine for you :-)

**Comment by MartinThoma** [comment]
Sorry that your contributions took me so long to notice :see_no_evil:

**Comment by hjoliver** [comment]
No worries, I've pushed a commit with your stylistic tweak.

**Pull request merged**